### ISSUE TITLE
Docker rootless - Permission issue?

### ISSUE THREAD

**Issue by Kirk1984** [nextcloud/all-in-one#823]
Hello,

today i tried setting up AiO (1.4.0) under Ubuntu 22.04 and ran into a permission issue. i setup a clean Debian 11 install with the same result. It is the same result with and without the "NEXTCLOUD_DATADIR" env variable.
The used compose file worked before during testing. I guess it started with 1.3.1, but sadly i didn't pay much attention because the old test microsd was corrupt.

This is the error the aio-nextcloud container puts out:
```
Configuring Redis as session handler
touch: /mnt/ncdata/this-is-a-test-file: Permission denied
The www-data user doesn't seem to have access rights in /mnt/ncdata.
Did you maybe change the datadir and did forget to apply the correct permissions?
```

And the Apache container waits for nextcloud:
```
nc: getaddrinfo for host "nextcloud-aio-nextcloud" port 9000: Name or service not known
Waiting for Nextcloud to start...
```

**Comment by szaimen** [comment]
Hi, can you show me the docker compose file?

Also did you follow https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir and adjusted the permissions correctly?

**Comment by Kirk1984** [comment]
My current docker-compose:
```
version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer

services:
  nextcloud:
    image: nextcloud/all-in-one:latest-arm64
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /$XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock:ro
    environment:
      - APACHE_PORT=11000
      - DOCKER_SOCKET_PATH=$XDG_RUNTIME_DIR/docker.sock
      - NEXTCLOUD_DATADIR=/mnt/ncdata
    ports:
      - 8080:8080
```

I also did the permissions you mentioned before starting the container. neither /mnt/ncdata nor any other location (a folder in /home for example) worked so far.

And i was wrong, it works without NEXTCLOUD_DATADIR, it just takes its sweet time on my Pi.

**Comment by szaimen** [comment]
> And i was wrong, it works without NEXTCLOUD_DATADIR, it just takes its sweet time on my Pi.

So then it works now and we can close this?

**Comment by szaimen** [comment]
> I also did the permissions you mentioned before starting the container. neither /mnt/ncdata nor any other location (a folder in /home for example) worked so far.

are you sure? Maybe it just takes a long time?

**Comment by szaimen** [comment]
If you've used `NEXTCLOUD_DATADIR=/mnt/ncdata` can you show me the currently applied permissions with `sudo ls -l /mnt/ncdata`?

**Comment by Kirk1984** [comment]
pi@ncaiotest:~ $ sudo ls -la /mnt/ncdata
total 8
drwxr-x--- 2 www-data root 4096 Jun 13 15:32 .
drwxr-xr-x 3 root     root 4096 Jun 13 15:32 ..

The nextcloud container just spams the error. 10 mins in and still only the error. 

But for science, i just started everything from the aio interface and i will go shopping. So far the error repeated itself 8 times. Will report back when i'm back with some Tofu :)

**Comment by szaimen** [comment]
> drwxr-x--- 2 www-data root 4096 Jun 13 15:32 .

Can you try to run `sudo chown -R 33:0 /mnt/ncdata` and `sudo chmod -R 750 /mnt/ncdata` again and see if it helps?

Also, can you post the output of `sudo docker inspect nextcloud-aio-nextcloud` here`
(obviously make sure to not post sensible information)

**Comment by Kirk1984** [comment]
Tofu is bought. The nextcloud and the Apache container (because it can't reach the nextcloud container) never started.

Reapplied the permissions, same outcome.

docker inspect nextcloud-aio-nextcloud output:

```
[
    {
        "Id": "3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7",
        "Created": "2022-06-13T14:54:59.362773985Z",
        "Path": "/start.sh",
        "Args": [
            "/usr/bin/supervisord",
            "-c",
            "/supervisord.conf"
        ],
        "State": {
            "Status": "restarting",
            "Running": true,
            "Paused": false,
            "Restarting": true,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 0,
            "ExitCode": 1,
            "Error": "",
            "StartedAt": "2022-06-13T14:56:22.99535343Z",
            "FinishedAt": "2022-06-13T14:56:23.09063451Z"
        },
        "Image": "sha256:64d53500970d406c354d0da8a7aa5c191c439dd391725c5ce7fe088324fe9233",
        "ResolvConfPath": "/home/pi/.local/share/docker/containers/3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7/resolv.conf",
        "HostnamePath": "/home/pi/.local/share/docker/containers/3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7/hostname",
        "HostsPath": "/home/pi/.local/share/docker/containers/3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7/hosts",
        "LogPath": "/home/pi/.local/share/docker/containers/3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7/3c932f50e28273035248162ca044a13ded3995175575807e6467b3a56d3f8fd7-json.log",
        "Name": "/nextcloud-aio-nextcloud",
        "RestartCount": 10,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": [
                "nextcloud_aio_nextcloud:/var/www/html:rw",
                "/mnt/ncdata:/mnt/ncdata:rw"
            ],
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": null,
            "RestartPolicy": {
                "Name": "unless-stopped",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "CgroupnsMode": "private",
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": false,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": null,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/home/pi/.local/share/docker/overlay2/4593ee33f2c1f304a7d16e9840200b93f8df94aaf681f512ed7697f25c825ea4-init/diff:/home/pi/.local/share/docker/overlay2/a679ea20dc46b18530e7b01dfc2f6fd5b5398ed733cc53b0ab7a3631f3071b47/diff:/home/pi/.local/share/docker/overlay2/d32acd8f283315244682fe6ee8bf05f7c0f0ddafdc5100803c1ccf1a46583e53/diff:/home/pi/.local/share/docker/overlay2/260f453c9c969db4bcae9ae3e295cac0d524293d48a970ae94e6c4b803751a18/diff:/home/pi/.local/share/docker/overlay2/072fd2fbf37f298aaf8bce9a599dfe625cc3065e50bb59be49ae7e69b3b601f7/diff:/home/pi/.local/share/docker/overlay2/7fac84a8292d9679722e3d158c3c42c0c40be1cfdd2bc1279ec51bb624d4c720/diff:/home/pi/.local/share/docker/overlay2/558e1130d7c78b7b9ca60c507e147dc9abc10aaf6c06911733382471bbba0e54/diff:/home/pi/.local/share/docker/overlay2/c7fc5248f717c26e32a1f57e510af3d4e701cdd9d5a22bfc812a0582a9d71c17/diff:/home/pi/.local/share/docker/overlay2/73d95fe8355574c5790a715552d5050d3dab40dc758140302e239d174053e2c2/diff:/home/pi/.local/share/docker/overlay2/fdaebad8db38cbdae478c7d5a697ac804be2a2c1c00b82a58d12a3d1e891851b/diff:/home/pi/.local/share/docker/overlay2/85e350626341ef0b7874e3e59904450a511188aad8ed09b8af12635d09ebfe44/diff:/home/pi/.local/share/docker/overlay2/196e2e25bfe45b670c69f081fea9f3792d2ff388b8d637687327b932e8ef7e3a/diff:/home/pi/.local/share/docker/overlay2/c596da6cdac533191cf327d1d407b59f0fff14740877c1fac1e192fd9d7a0557/diff:/home/pi/.local/share/docker/overlay2/267e5f6f113dbd52c3d77b752497b4fba4abef094b9218058b9bd63fe6a71bfa/diff:/home/pi/.local/share/docker/overlay2/7f8c6cc885af6ae452bdc933db5e3c2b755106df44cdb80ca6767cf209e67b88/diff:/home/pi/.local/share/docker/overlay2/8ddfc164cf29ae7932c3bcce990904311998cd1e38e11e17269b4518533594d1/diff:/home/pi/.local/share/docker/overlay2/b014c4afe9e1de91ce78ea8beee49fae834b294285d6997d7ccbba6727e4c649/diff:/home/pi/.local/share/docker/overlay2/3ca199875f6cbfbea03f972016c02c359ff5e95e845fab9d04f716ce2a9d0b18/diff:/home/pi/.local/share/docker/overlay2/62170ea311d7d05adf2acc034551aacf6db0f8711a4ac19bf23513772474e213/diff:/home/pi/.local/share/docker/overlay2/b040b9e6180a9cf17240347ffa253ee2e27d5f8635018447669adb23035a7df6/diff:/home/pi/.local/share/docker/overlay2/50cf7756cdd152edf0ae914f7495d6de7b5957cdc9e4b59c9f24b465a0221592/diff:/home/pi/.local/share/docker/overlay2/27d6a8adafd6a49a785e04ce321c3e61ef29f0e22f3e649952263d8e9053b49a/diff:/home/pi/.local/share/docker/overlay2/4f02efd489df6d6526f20dee535ff582a34c1513c9ca419322103e1348fb4bf4/diff:/home/pi/.local/share/docker/overlay2/963ff70c9eb7ac6edf4553029f8be5421914ec3c6a602a2c4634b293fd729259/diff:/home/pi/.local/share/docker/overlay2/effcda1ad1cbbae4a6ddd02e217cb09d3204b8cd37ff6c2e93edca75395d87ab/diff:/home/pi/.local/share/docker/overlay2/25c55740f369b1e3a3f2bb8c692e827f022e7e3868d1fb3effd842688e0ff4a6/diff:/home/pi/.local/share/docker/overlay2/d93d600f1b5562d17f40d54a1f6688e3a996d72cb1cf808fc0ee1c9d2520dd1e/diff:/home/pi/.local/share/docker/overlay2/6d9dbfac2a092f1e6098176413b88e20caa2d62a8ebdd432642ca1bfcb10ad81/diff:/home/pi/.local/share/docker/overlay2/df71f32a1cc24c2b6c7f7cc59555e4df728f248f748d7598be1307088a80619e/diff:/home/pi/.local/share/docker/overlay2/b9dd61c4a129a1afeeec26b8ff1a75c581bd734f16c9eea6c958934f31e08e8c/diff:/home/pi/.local/share/docker/overlay2/ebe2077ff6335602e87da7438e7c7099ea6492d8e14f322c93f2d6337d40cc65/diff",
                "MergedDir": "/home/pi/.local/share/docker/overlay2/4593ee33f2c1f304a7d16e9840200b93f8df94aaf681f512ed7697f25c825ea4/merged",
                "UpperDir": "/home/pi/.local/share/docker/overlay2/4593ee33f2c1f304a7d16e9840200b93f8df94aaf681f512ed7697f25c825ea4/diff",
                "WorkDir": "/home/pi/.local/share/docker/overlay2/4593ee33f2c1f304a7d16e9840200b93f8df94aaf681f512ed7697f25c825ea4/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [
            {
                "Type": "volume",
                "Name": "nextcloud_aio_nextcloud",
                "Source": "/home/pi/.local/share/docker/volumes/nextcloud_aio_nextcloud/_data",
                "Destination": "/var/www/html",
                "Driver": "local",
                "Mode": "rw",
                "RW": true,
                "Propagation": ""
            },
            {
                "Type": "bind",
                "Source": "/mnt/ncdata",
                "Destination": "/mnt/ncdata",
                "Mode": "rw",
                "RW": true,
                "Propagation": "rprivate"
            }
        ],
        "Config": {
            "Hostname": "3c932f50e282",
            "Domainname": "",
            "User": "www-data",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "9000/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "POSTGRES_HOST=nextcloud-aio-database",
                "POSTGRES_PASSWORD=4c9068a9b2710fd510f1e7aaaf2ad0f5184be6cf8b3744bc",
                "POSTGRES_DB=nextcloud_database",
                "POSTGRES_USER=nextcloud",
                "REDIS_HOST=nextcloud-aio-redis",
                "REDIS_HOST_PASSWORD=0f1ef992bf52634e007d9272e058e44b1d0a5da44bc9aa8f",
                "AIO_TOKEN=c1c8d41f960ad1ef9114bcfff0c0251351aa61948d97a4ee",
                "NC_DOMAIN=sensiblestuff.spdns.de",
                "ADMIN_USER=admin",
                "ADMIN_PASSWORD=aa33e7291f9somethinge7d789c365c0607fbbeb5fa",
                "NEXTCLOUD_DATA_DIR=/mnt/ncdata",
                "OVERWRITEHOST=sensiblestuff.spdns.de",
                "OVERWRITEPROTOCOL=https",
                "TURN_SECRET=91b8525b2416somethingda658307042c2e13522",
                "SIGNALING_SECRET=318172985e3088something9cd7982a2a61439210cba9df",
                "AIO_URL=192.168.168.141:8080",
                "NEXTCLOUD_MOUNT=",
                "CLAMAV_ENABLED=",
                "CLAMAV_HOST=nextcloud-aio-clamav",
                "ONLYOFFICE_ENABLED=",
                "COLLABORA_ENABLED=yes",
                "COLLABORA_HOST=nextcloud-aio-collabora",
                "TALK_ENABLED=yes",
                "ONLYOFFICE_HOST=nextcloud-aio-onlyoffice",
                "DAILY_BACKUP_RUNNING=",
                "TZ=Europe/Berlin",
                "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PHPIZE_DEPS=autoconf \t\tdpkg-dev dpkg \t\tfile \t\tg++ \t\tgcc \t\tlibc-dev \t\tmake \t\tpkgconf \t\tre2c",
                "PHP_INI_DIR=/usr/local/etc/php",
                "PHP_CFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_CPPFLAGS=-fstack-protector-strong -fpic -fpie -O2 -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64",
                "PHP_LDFLAGS=-Wl,-O1 -pie",
                "GPG_KEYS=1729F83938DA44E27BA0F4D3DBDB397470D12172 BFDDD28642824F8118EF77909B67A5C12229118F",
                "PHP_VERSION=8.0.19",
                "PHP_URL=https://www.php.net/distributions/php-8.0.19.tar.xz",
                "PHP_ASC_URL=https://www.php.net/distributions/php-8.0.19.tar.xz.asc",
                "PHP_SHA256=ba62219c4b0486cbb2a04f0796749a46b0ee1f5a142ed454212b4e2460cb0fab",
                "PHP_MEMORY_LIMIT=512M",
                "PHP_UPLOAD_LIMIT=10G",
                "NEXTCLOUD_VERSION=23.0.5",
                "NEXTCLOUD_UPDATE=1"
            ],
            "Cmd": [
                "/usr/bin/supervisord",
                "-c",
                "/supervisord.conf"
            ],
            "Image": "nextcloud/aio-nextcloud:latest-arm64",
            "Volumes": {
                "/mnt/ncdata": {},
                "/var/www/html": {}
            },
            "WorkingDir": "/var/www/html",
            "Entrypoint": [
                "/start.sh"
            ],
            "OnBuild": null,
            "Labels": {},
            "StopSignal": "SIGQUIT"
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "5d153a88c985b38385b9c7bc4ddfd1f98b49e4c90fbefdaeed489418cda297ad",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {},
            "SandboxKey": "/run/user/1000/docker/netns/5d153a88c985",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "",
            "Gateway": "",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "",
            "IPPrefixLen": 0,
            "IPv6Gateway": "",
            "MacAddress": "",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "e6d8fc33e0dcacc226423a360813df3df88ce5aa324c45f91db621a299bcc10d",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                },
                "nextcloud-aio": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": [
                        "3c932f50e282"
                    ],
                    "NetworkID": "09cbe1600ac5b673a303c6d97057f3d0e347690bb83b5f12c45a20ce0a12cae8",
                    "EndpointID": "",
                    "Gateway": "",
                    "IPAddress": "",
                    "IPPrefixLen": 0,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "",
                    "DriverOpts": null
                }
            }
        }
    }
]

```

**Comment by 2020Sanoj** [comment]
Can I also try to install it again on another server and test again?

**Comment by Kirk1984** [comment]
Later i will setup docker on that system regulary and see if it still happens when docker runs as root.

**Comment by szaimen** [comment]
> Later i will setup docker on that system regulary and see if it still happens when docker runs as root.

All right, thanks!

**Comment by Kirk1984** [comment]
It runs with "normal" docker. i even used the /mnt/ncdata fodler i created for the rootless setup without any issues and i can confirm that the files land are stored there.

I can't get where this is coming from. I have 2 more Boxes with rootless docker (same docker version) and several container (caddy, calibre-web, nextcloudpi, homer, vaultwarden, etc) that all access files in folders. but in these cases they are specified as volumes (caddy example: ~/caddy/data:/data).

**Comment by szaimen** [comment]
I also do not understand why this happens... Maybe can you try to adjust the permissions to 777 for a test so that we can see with which user subfolders get created?

**Comment by szaimen** [comment]
hm... maybe the problem is that no other user apart from root can access /mnt itself?

**Comment by szaimen** [comment]
I guess I did read the permissions incorrectly... so still no clue why this does not work... 777 should definitely make it work. I guess the container needs to be recreated to make a real test...

**Comment by Kirk1984** [comment]
I always deleted the container and volumes. Switching from rootless to root and back i also redownloaded the images. Every test was from the ground up.

**Comment by szaimen** [comment]
I wonder why the volume seems to work but the host mount not... strange :/

**Comment by Kirk1984** [comment]
I don't get it either. I had few things with docker now and then, but this is a new one. Tomorrow i try again with a home folder and make sure to follow the corrent permission procedure. Hopefully something will happen.

**Comment by Kirk1984** [comment]
So, i got it running with rootless docker.

I recreated everything without a specific datadir and checked the permissions for the volumes. 100032:100032. when i applied those to "/home/pi/ncaio/ncdata" nextcloud started normally. I copied a testfile to my instance and can confirm that it is in the folder "/home/pi/ncaio/ncdata/admin/files"). 
Then i tried a Backup with "/mnt/backup" as the destination. that Folder was created with "sudo mkdir /mnt/backup". The Backup failed so i applied 100032:100032 to "/mnt/backup" and the Backup worked.

could it be that, because of some reason, that the Folder permission must be in between these values?:
```
pi@ncaiotest:~ $ grep ^$(whoami): /etc/subuid
pi:100000:65536
pi@ncaiotest:~ $ grep ^$(whoami): /etc/subgid
pi:100000:65536
```

Because when i initially tested the Backup  it worked without adjusting permissions on "/mnt/backup". 
[https://github.com/nextcloud/all-in-one/issues/558#issuecomment-1126184181](url)

**Comment by szaimen** [comment]
> 100032 

Ia this the id of the user that runs the docker daemon?

If yes I wonder why you have so different test results in comparison to before when you tried it out a few weeks ago? One possible explanation is that you've made a mistake when you've started aio the first time so that it was using the root docker socket...

To me it now looks like how I would have expected it to work from the beginning... I guess we need to document this then in the docker rootles documentation... WDYT?

**Comment by szaimen** [comment]
Can you state which user is running the docker daemon in the above cases?

**Comment by Kirk1984** [comment]
I can check that tomorrow, but all setups are created like docker documents it for the rootless setup. So pi and ubuntu i guess. The default users on the preconfigured images.

**Comment by szaimen** [comment]
But then it works as expected, no?

So all files outside the containers get created as the user that is running the docker daemon...

**Comment by Kirk1984** [comment]
My guess is that its a Bug. Tomorrow i will test it on a new VM, with alls the steps i did a gajilion times by now. If it works, it must be an issue with docker on arm64. If not, i'll go mad :)

**Comment by Kirk1984** [comment]
I set up a fresh debian 11 vm and the issue remains. I'm at a loss. So far i have one box where ist works as it always did and 3 where it doesn't.

But i was wrong, the one box where it works is debian 10, not 11. 

I will try again with a debian 10 install later.

**Comment by Kirk1984** [comment]
I just checked on my oldest machine running docker. All Folders in /home that i linked to containers are 100999:100999. Either i created them and linked them or docker created them. But all Folders are 100999:100999. And that was set by docker, be it created by my user or created when i ran docker-compose (the debian 10 has the old docker-compose, not the new docker compose).

**Comment by szaimen** [comment]
> Could it be that docker compose does something different than docker-compose when running rootless?

Honestly I don't know... I have not really used docker rootless at all...

**Comment by szaimen** [comment]
My only guess is that if it works on the new machines repeatedley like this I do not think it is a bug in the new version... Rather a bug in the old version was fixed... And it is not that bad since we can simply document the change

**Comment by Kirk1984** [comment]
Maybe you can spin up a vm. So far i am the only one testing it (wasn't there someone who wanted to test it too?). Maybe i'm stupid. According to my wife, that is a distinct possibility :)

**Comment by Kirk1984** [comment]
I just tested it with "docker run -d" instead of docker compose (homer). Same outcome.

If there was a change it isn't very well documented by the docker guys.

**Comment by szaimen** [comment]
> Maybe i'm stupid.

I don't think you are stupid. As I stated above it now works like I would have expected it to work from the beginning. I will document this in the docker rootles documentation.

**Comment by Kirk1984** [comment]
So, this is going to be long and confusing. at least for me. Some common things i did:

All folders used in the compose files where created with the user pi, all have pi:pi as the owner. Just plain "mkdir folder" as pi, no special sauce.
I set up docker and docker rootless according to the docker docs. with the repo installed.
OS is RaspberryOS (Debian 11)

i testet 3 Apps via compose file:
homer
nextcloud All in One
nextcloud from linuxserver.io

tl;dr: nextcloud from linuxserver.io works. And it works after a reboot. There must be something different with that image.

homer compose:
```
version: "2"
services:
  homer:
    image: b4bz/homer:latest
    #To build from source, comment previous line and uncomment below
    #build: .
    container_name: homer
    volumes:
      - /home/pi/homer/assets:/www/assests
    ports:
      - 6789:8080
    user: 1000:1000
    environment:
      - INIT_ASSETS=1
    restart: unless-stopped
```
Result: No permission to the folder with pi:pi. 


Nextcloud All in One compose:
```
version: "3.8"

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer

services:
  nextcloud:
    image: nextcloud/all-in-one:latest-arm64
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      - /$XDG_RUNTIME_DIR/docker.sock:/var/run/docker.sock:ro
#      - /var/run/docker.sock:/var/run/docker.sock:ro
    environment:
      - APACHE_PORT=11000
      - DOCKER_SOCKET_PATH=$XDG_RUNTIME_DIR/docker.sock
      - NEXTCLOUD_DATADIR=/home/pi/ncaio/ncdata
    ports:
      - 8080:8080
 ```
Result: No permission to the folder with pi:pi. I then applied the chown/chmod commands documented for Nextcloud All in One, still the same, no permission.


nextcloud compose (linuxserver.io):
```
version: "2.1"
services:
  nextcloud:
    image: lscr.io/linuxserver/nextcloud:latest
    container_name: nextcloud
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=Europe/London        #forgot to change the timezone :)
    volumes:
      - /home/pi/linuxsercernc/appdata:/config
      - /home/pi/linuxservernc/data:/data
    ports:
      - 4430:443
    restart: unless-stopped
```
Result: Works. The /home/pi/linuxservernc/appdata Folder is empty and pi:pi. The /home/pi/linuxservernc/data is filled and i can confirm that a testfile i copied throught Nextcloud via drag and drop into the Browser is in there. But that folder is 100999:100999. I dind't set that. that was the container:
```
drwxr-xr-x 2 pi     pi     4096 Jun 15 18:44 appdata
-rw-r--r-- 1 pi     pi      352 Jun 15 18:45 docker-compose.yml
drwxrwx--- 5 100999 100999 4096 Jun 15 19:38 data
```
According to their docs [https://docs.linuxserver.io/general/running-our-containers](url) the /config seems to be in every container from them. Maybe it just isn't used in the nextcloud container. because the Folder in is empty and everything works so far.

I don't get it. Homer and Nextcloud All in One give me permission errors and can't access the Folder. the nextcloud linuxserver.io image seems to chown it and it works.

Sorry for the Wall of text. I wanted to test some images to make sure that it is a docker issue. But the first i tired worked, makign this even more confusing for me.

**Comment by Kirk1984** [comment]
I forgot to do something...

I applied 100999:100999 to the Nexcloud All in One Folder (NEXTCLOUD_DATADIR=/home/pi/ncaio/ncdata).

```
-rw-r--r-- 1 pi     pi      611 Jun 14 06:36 docker-compose.yml
drwxr-x--- 2 100999 100999 4096 Jun 15 19:09 ncdata
```
Sadly, it doesn't work. It still spams the log about permissions.

**Comment by szaimen** [comment]
I've made it work on my teste instance. Basically it looks like the nextcloud container is running as the 100032 user and e.g. the borg container is indeed running as the user that is running the docker daemon since the process inside is running as root user.

So to conclude: if the process inside a  docker container is by default not running as the root user but as a different user, it will most likely use the id 100032. if the process inside a docker container is running as the root user, it will use the id of the user that is running the docker daemon.

**Comment by Kirk1984** [comment]
I will test that tomorrow. Thank you for looking into it!

**Comment by szaimen** [comment]
Interestingly is e.g. the database container having another uid inside because probably it is using a different user inside the container...

in conclusion it seems like the www-data user inside a rootless container is in general translated into the uid 100032.

**Comment by szaimen** [comment]
I see now how this works:
`grep ^$(whoami): /etc/subuid` shows 1000 as the first value.

The process inside the apache and nextcloud container are running with the id 33. The first usable id for user containers is 1000. The userid that you get is then basically 1000 concat 33 - 1 = 100032. This also applies to the database container which usses internally the id 999 which then results in 1000998.

**Comment by szaimen** [comment]
I've adjusted the documentation accordingly in https://github.com/nextcloud/all-in-one/pull/829

**Comment by Kirk1984** [comment]
Yes, it works. I didn't even have to recreate anything. Just stop everything, chown it to 100032:100032 and start it again.

But why can't that be applied by the container?

**Comment by szaimen** [comment]
> But why can't that be applied by the container?

Because not process inside the nextcloud container is running as root user.

**Comment by szaimen** [comment]
Anyway, the documentation works now so this is done.

**Comment by Kirk1984** [comment]
> > But why can't that be applied by the container?
> 
> Because no process inside the nextcloud container is running as root user.

Ah ok. That makes sense.
